apps/opt.c

```diff
@@ -656,22 +659,54 @@ int opt_next(void)
             /* Just a string. */
             break;
         case '/':
-            if (app_isdir(arg) >= 0)
+            if (app_isdir(arg) > 0)
                 break;
             BIO_printf(bio_err, "%s: Not a directory: %s\n", prog, arg);
             return -1;
         case '<':
             /* Input file. */
-            if (strcmp(arg, "-") == 0 || app_access(arg, R_OK) >= 0)
+            if (strcmp(arg, "-") == 0 || app_access(arg, R_OK) == 0)
                 break;
             BIO_printf(bio_err,
                        "%s: Cannot open input file %s, %s\n",
                        prog, arg, strerror(errno));
             return -1;
         case '>':
             /* Output file. */
-            if (strcmp(arg, "-") == 0 || app_access(arg, W_OK) >= 0 || errno == ENOENT)
+            l = strlen(arg);
+            if (arg[l - 1] == '/' || app_isdir(arg) > 0) {
```

Comment:
Path string like "/tmp/non-exist/" or "/tmp/non-exist" is passed, `app_isdir` will return < -1 for both of them. But the former one (with the trailing slash) should not be processed any more, that is because the following `check-parent-writable' logic will judge this as a 'good' (`/tmp` is writable). For simplicity, just kick out all the strings trailing with a 'slash'. It seems it's valid for the `-out` option.